super-pane/use-paginated-client.ts

```diff
@@ -46,6 +49,13 @@ function usePaginatedClient({ typeName, pageSize, selectedColumns }: Params) {
   const [refreshId, setRefreshId] = useState(nanoid());
   const refresh = useCallback(() => setRefreshId(nanoid()), []);
 
+  const [userQuery, setUserQuery] = useState('')
+  // Builds the string to use when a custom filter has been entered
+  const buildUserQuery = useCallback(() => {
+    return userQuery.length ? ` && ${selectedFieldToSearchFor.name} match "${userQuery}*"` : ''
+  }, [selectedFieldToSearchFor, userQuery])
```

Comment:
The reason it's not a string at the moment, is that it would have had to be duplicated in all the three useEffects, and these in turn would have had to have userQuery and selectedFieldToSearchFor in their deps array to stay reactive. But another solution would be to save it as a string creating a useEffect that has userQuery and selectedFieldToSearchFor in its deps array, and then saving the string to a useState hook, and then have the three useEffects depend on that string?

Reply:
> The reason it's not a string at the moment, is that it would have had to be duplicated in all the three useEffects, and these in turn would have had to have userQuery and selectedFieldToSearchFor in their deps array to stay reactive

i guess i'm just saying i don't think it needs to be a callback. like wrapping it in a useMemo would have the same effect right?

```diff
- const buildUserQuery = useCallback(() => {
+ const searchQuery = useMemo(() => {
  return userQuery.length ? ` && ${selectedFieldToSearchFor.name} match "${userQuery}*"` : ''
}, [selectedFieldToSearchFor, userQuery])
```

the memoed version,`searchQuery`, still has the deps on `selectedFieldToSearchFor` and `userQuery` now without requiring the wrapping function.

then to take things one step further, i don't think even the `useMemo` is needed since the computation returns a primitive/literal type (i.e. a string), which is compared literally in React (e.g. `'a' === 'a'`).

So this should be fine:

```diff
- const searchQuery = useMemo(() => {
-   return userQuery.length ? ` && ${selectedFieldToSearchFor.name} match "${userQuery}*"` : ''
- }, [selectedFieldToSearchFor, userQuery])
+ const searchQuery = userQuery.length ? ` && ${selectedFieldToSearchFor.name} match "${userQuery}*"` : ''
```

downstream, i think `searchQuery` can replace `buildUserQuery` in all three effect and everything should still be reactive. let me know if i missed something though!

this is just a small knit though. it definitely works as-is so i'll resolve so we can merge